src/Daemon/BaseDaemon.cpp

```diff
@@ -664,6 +664,10 @@ void BaseDaemon::initialize(Application & self)
 
     DB::ConfigProcessor(config_path).savePreprocessedConfig(loaded_config, "");
 
+#if USE_SSL
+    DB::ConfigProcessor(config_path).decryptConfig(loaded_config);
```

Comment:
Not sure but since `ConfigReloader::reloadIfNewer` does the decryption by itself: can we move the decrypt call on l. 668 into some existing method of `ConfigProcessor`, e.g. `savePreprocessedConfig` (l. 665).

Reply:
Yes. Probably it's a good to move it to `savePreprocessedConfig`

But the problem is that `loaded_config` is `const`:

```c++
void ConfigProcessor::savePreprocessedConfig(const LoadedConfig & loaded_config, std::string preprocessed_dir)
```

Reply:
OK. I removed `const` in `savePreprocessedConfig` and moved decryption code there.